File: data/readable_format/cpp/file1339.cpp
#include <gtest/gtest.h>
#include <string>

// Definition of the Injection struct
struct Injection {
    // Double precision variables
    double RCM_injection = 0.0;
    double ZCM_injection = 0.0;
    double Rwidth_injection = 0.0;
    double Zwidth_injection = 0.0;
    double XCM_injection = 0.0;
    double YCM_injection = 0.0;
    double Xwidth_injection = 0.0;
    double Ywidth_injection = 0.0;
    double alpha0 = 0.0;
    double beta0 = 0.0;
    double V0 = 0.0;
    double devalpha = 0.0;
    double devbeta = 0.0;
    double devV = 0.0;

    // Character variables
    std::string inject_type_pos = "pointlike";
    std::string inject_type_vel = "stationary";

    // Logical variables
    bool inject_coord_cart = false;
    bool inject_coord_cyl = false;
    bool inject_coord_tilt = false;
    bool rand_seed = true;
};

// Test Fixture for Injection struct
class InjectionTest : public ::testing::Test {
protected:
    Injection injection; // Test fixture for the Injection struct
};

// Test case for Initialization
TEST_F(InjectionTest, Initialization) {
    // Test double initializations
    EXPECT_DOUBLE_EQ(0.0, injection.RCM_injection);
    EXPECT_DOUBLE_EQ(0.0, injection.ZCM_injection);
    EXPECT_DOUBLE_EQ(0.0, injection.alpha0);

    // Test string initializations
    EXPECT_EQ("pointlike", injection.inject_type_pos);
    EXPECT_EQ("stationary", injection.inject_type_vel);

    // Test boolean initializations
    EXPECT_FALSE(injection.inject_coord_cart);
    EXPECT_TRUE(injection.rand_seed);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/readable_format/fortran/file1339.f90
MODULE injection
  DOUBLE PRECISION    :: RCM_injection=0., ZCM_injection=0., Rwidth_injection=0., Zwidth_injection=0., &
                         XCM_injection=0., YCM_injection=0., Xwidth_injection=0., Ywidth_injection=0., &
                         alpha0 = 0., beta0 = 0., V0 = 0., devalpha = 0., devbeta = 0., devV = 0.
  CHARACTER(LEN=10)   :: inject_type_pos='pointlike', inject_type_vel='stationary'
  LOGICAL             :: inject_coord_cart=.FALSE., inject_coord_cyl=.FALSE., inject_coord_tilt=.FALSE., rand_seed = .TRUE.
END MODULE injection

PROGRAM test_injection
  USE injection
  IMPLICIT NONE

  ! Test the initialization of module variables
  IF (RCM_injection /= 0.0) PRINT *, 'Test failed for RCM_injection'
  IF (ZCM_injection /= 0.0) PRINT *, 'Test failed for ZCM_injection'
  IF (alpha0 /= 0.0) PRINT *, 'Test failed for alpha0'
  IF (TRIM(inject_type_pos) /= 'pointlike') PRINT *, 'Test failed for inject_type_pos'
  IF (inject_coord_cart .NEQV. .FALSE.) PRINT *, 'Test failed for inject_coord_cart'
  IF (inject_coord_cyl .NEQV. .FALSE.) PRINT *, 'Test failed for inject_coord_cyl'
  IF (inject_coord_tilt .NEQV. .FALSE.) PRINT *, 'Test failed for inject_coord_tilt'
  IF (rand_seed .NEQV. .TRUE.) PRINT *, 'Test failed for rand_seed'
  PRINT *, 'All tests passed.'

END PROGRAM test_injection